src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -311,17 +328,31 @@ def hash_supported(self, algorithm):
         return evp_md != self._ffi.NULL
 
     def scrypt_supported(self):
-        return self._lib.Cryptography_HAS_SCRYPT == 1
+        if self._fips_enabled:
+            return False
```

Comment:
Insert salty comment here since I definitely disagree with OpenSSL 3.0.0's position on this.